### DISCUSSION TITLE
Episode 116: Max Gap Count in C++23

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#6] [Episode]
Episode Link: https://adspthepodcast.com/2023/02/10/Episode-116.html

Feel free to write comments or ask questions 🙂

**Comment by kentf** [comment]
What was the recipe for the Cathay Pacific lounge noodle soup :)?

**Reply by codereport** [reply]
@brycelelbach, this one is for you : )

**Reply by brycelelbach** [reply]
[Here's the receipe](https://goodmagazine.co.nz/bring-the-taste-of-asia-into-your-home/)!

**Reply by kentf** [reply]
Thank you for the recipe!

It sure looks delicious based on your description and the steps in the recipe!

However, it was almost a complete disaster! 😄 

The soup part was almost inedible. I'm not quite sure where it went wrong, but I think the sesame paste I used was too bitter/flavoury.

Since I separated the chicken meat and made a stock only the bones, the meal was saved by sauteing chicken and mixing with noodles.

Next time I'll order the food from the professionals 👍

**Comment by Mike278** [comment]
Cool episode! I don't know any array languages, but the way you described the array language solution at 10:11+ made me think of how I'd solve it in SQL - basically count how many diffs equal the max diff.

```sql
create table input (x);
insert into input values (5), (3), (1), (10), (20), (15);

with diffs(diff) as (
    select x - lag(x, 1) over (order by x)
    from input
)
select count(*)
from diffs
where diff = (select max(diff) from diffs);
```
```
$ cat adsp116.sql | sqlite3 :memory:
3
```
